src/components/ui4/BlogCards.tsx

```diff
@@ -0,0 +1,44 @@
+import { BlogPost, BlogPostGeo } from 'edge-info-server'
+import React from 'react'
+import { ListRenderItem } from 'react-native'
+import { useSafeAreaFrame } from 'react-native-safe-area-context'
+
+import { useHandler } from '../../hooks/useHandler'
+import { infoServerData } from '../../util/network'
+import { useTheme } from '../services/ThemeContext'
+import { BlogCard } from './BlogCard'
+import { CarouselUi4 } from './CarouselUi4'
+
+export interface Props {
+  countryCode?: string
+}
+
+export const BlogCards = (props: Props) => {
+  const { countryCode } = props
+  const { width } = useSafeAreaFrame()
+  const theme = useTheme()
+
+  const [blogCards, setBlogCards] = React.useState<BlogPost[]>([])
+
+  const renderBlog: ListRenderItem<BlogPost> = useHandler(({ item }) => <BlogCard blogPost={item} />)
+
+  // Check for BlogCards from info server:
+  React.useEffect(() => {
+    if (countryCode == null) return
+
+    const filteredBlogPostsGeo = (infoServerData.rollup?.blogPostsGeo ?? []).filter((blog: BlogPostGeo) => {
```

Comment:
Pull out all the filtering logic into a unit testable routine and write unit tests for them. See `validatePromoCardsInner` as an example.

This logic will likely grow to include more conditions in the future and we want to start it with clean test driven development.